llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -263,14 +266,22 @@ class PipelineExtractor : public PipelineScheduleVisitor {
     // Prologue and epilogue obtain copies.
     MachineInstr *ToBeEmitted =
         InLoop ? MI : Loop.TheBlock->getParent()->CloneMachineInstr(MI);
```

Comment:
I think we should prefer not using `CloneMachineInstr`, since this is not added to MBB and other structure like `BlockState`

Reply:
However, using CloneMachineInstr is essentially what the post pipeliner needs. When we call this extraction, everything in the schedule has been frozen. We can materialize the pseudos just before calling the extractor.

Reply:
If you see a better way to emit an exact copy of an instruction into another basic block, please let me know.

Reply:
We had discussed that offline with Krishnam. The post-pipeliner should only deal with "finalized" instructions when materializing the schedule, i.e. no multi-slot opcodes anywhere. So cloning is fine at this stage, because state like `AlternateOpcodes` is outdated (and using it should be forbidden with asserts if possible).